src/TwitchLib.Communication/Clients/ClientBase.cs

```diff
@@ -1,369 +1,384 @@
-        {
-            Logger?.TraceMethodCall(GetType());
-            if (Token.IsCancellationRequested)
-            {
-                return;
-            }
+        if (OnSendFailed != null) await OnSendFailed.Invoke(this, eventArgs);
+    }
 
-            if(OnSendFailed != null) await OnSendFailed.Invoke(this, eventArgs);
+    /// <summary>
+    ///     Wont raise the given <see cref="EventArgs"/> if <see cref="Token"/>.IsCancellationRequested
+    /// </summary>
+    internal async Task RaiseError(OnErrorEventArgs eventArgs)
+    {
+        Logger?.TraceMethodCall(GetType());
+        if (Token.IsCancellationRequested)
+        {
+            return;
         }
 
-        /// <summary>
-        ///     Wont raise the given <see cref="EventArgs"/> if <see cref="Token"/>.IsCancellationRequested
-        /// </summary>
-        internal async Task RaiseError(OnErrorEventArgs eventArgs)
-        {
-            Logger?.TraceMethodCall(GetType());
-            if (Token.IsCancellationRequested)
-            {
-                return;
-            }
+        if (OnError != null) await OnError.Invoke(this, eventArgs);
+    }
 
-            if (OnError != null) await OnError.Invoke(this, eventArgs);
+    /// <summary>
+    ///     Wont raise the given <see cref="EventArgs"/> if <see cref="Token"/>.IsCancellationRequested
+    /// </summary>
+    private async Task RaiseReconnected()
+    {
+        Logger?.TraceMethodCall(GetType());
+        if (Token.IsCancellationRequested)
+        {
+            return;
         }
 
-        /// <summary>
-        ///     Wont raise the given <see cref="EventArgs"/> if <see cref="Token"/>.IsCancellationRequested
-        /// </summary>
-        private async Task RaiseReconnected()
-        {
-            Logger?.TraceMethodCall(GetType());
-            if (Token.IsCancellationRequested)
-            {
-                return;
-            }
+        if (OnReconnected != null) await OnReconnected.Invoke(this, new OnConnectedEventArgs());
+    }
 
-            if (OnReconnected != null) await OnReconnected.Invoke(this, new OnConnectedEventArgs());
+    /// <summary>
+    ///     Wont raise the given <see cref="EventArgs"/> if <see cref="Token"/>.IsCancellationRequested
+    /// </summary>
+    internal async Task RaiseMessage(OnMessageEventArgs eventArgs)
+    {
+        Logger?.TraceMethodCall(GetType());
+        if (Token.IsCancellationRequested)
+        {
+            return;
         }
 
-        /// <summary>
-        ///     Wont raise the given <see cref="EventArgs"/> if <see cref="Token"/>.IsCancellationRequested
-        /// </summary>
-        internal async Task RaiseMessage(OnMessageEventArgs eventArgs)
-        {
-            Logger?.TraceMethodCall(GetType());
-            if (Token.IsCancellationRequested)
-            {
-                return;
-            }
+        if (OnMessage != null) await OnMessage.Invoke(this, eventArgs);
```

Comment:
out of curiosity, could this be a null conditional? not a blocker, just wondering if im missing something.

Reply:
you are right we can't use null conditiona because of `await`